Error when executing the icf-provider script in Streaming Integrator.

### Description

We are observing an error when executing the "icf-provider" script which is related to the Streaming Integrator-4.2.0 version. 

### Steps to Reproduce

Convert the activemq-client-5.x.x.jar using the icf-provider script as mentioned in this document[1].

[1] - https://apim.docs.wso2.com/en/latest/use-cases/examples/streaming-examples/receive-jms-in-json-format/#note

### Affected Component

SI

### Version

4.2.0

### Environment Details (with versions)

OS - Mac OS 13.5.1 (22G90)
JDK - openjdk version "11.0.19

### Relevant Log Output

```shell
JAVA_HOME environment variable is set to /Users/gayanthanamarathna/Documents/jdk-11.0.19+7/Contents/Home CARBON_HOME environment variable is set to /Users/gayanthanamarathna/Documents/support/Setup/wso2si-tooling-4.2.0 ../../activemq-client-5.9.0/internal/CustomBundleActivator.java:2: error: package org.osgi.framework does not exist import org.osgi.framework.BundleActivator;                          ^ ../../activemq-client-5.9.0/internal/CustomBundleActivator.java:3: error: package org.osgi.framework does not exist import org.osgi.framework.BundleContext;                          ^ ../../activemq-client-5.9.0/internal/CustomBundleActivator.java:8: error: cannot find symbol public class CustomBundleActivator implements BundleActivator {                                               ^   symbol: class BundleActivator ../../activemq-client-5.9.0/internal/CustomBundleActivator.java:11: error: cannot find symbol     public void start(BundleContext bundleContext) throws Exception {                       ^   symbol:   class BundleContext   location: class CustomBundleActivator ../../activemq-client-5.9.0/internal/CustomBundleActivator.java:16: error: cannot find symbol     public void stop(BundleContext bundleContext) throws Exception {                      ^   symbol:   class BundleContext   location: class CustomBundleActivator ../../activemq-client-5.9.0/internal/CustomBundleActivator.java:10: error: method does not override or implement a method from a supertype     @Override     ^ ../../activemq-client-5.9.0/internal/CustomBundleActivator.java:15: error: method does not override or implement a method from a supertype     @Override     ^ 7 errors Aug 30, 2023 8:11:45 PM org.wso2.carbon.tools.spi.ICFProviderTool execute INFO: Executing 'jar uf ../../activemq-client-5.9.0/activemq-client-5.9.0.jar -C ../../activemq-client-5.9.0 /internal/CustomBundleActivator.class' Aug 30, 2023 8:11:45 PM org.wso2.carbon.tools.spi.ICFProviderTool addBundleActivatorHeader INFO: Running jar to bundle conversion Aug 30, 2023 8:11:45 PM org.wso2.carbon.tools.converter.utils.BundleGeneratorUtils convertFromJarToBundle INFO: Created the OSGi bundle activemq_client_5.9.0_1.0.0.jar for JAR file ../../activemq-client-5.9.0/activemq-client-5.9.0.jar
```


### Related Issues

_No response_

### Suggested Labels

product/si

--- a/tools/tools-core/src/main/java/org/wso2/carbon/tools/Constants.java
+++ b/tools/tools-core/src/main/java/org/wso2/carbon/tools/Constants.java
@@ -44,6 +44,8 @@ public class Constants {
     public static final String JAR_FILE_EXTENSION = ".jar";
     public static final String ZIP_FILE_EXTENSION = ".zip";
 
+    public static final String ECLIPSE_OSGI_VERSION = "org.eclipse.osgi_3.14.0.v20190517-1309.jar";
+
     //  create zip file system properties
     public static final String CREATE_NEW_ZIP_FILE_PROPERTY = "create";
     public static final String ENCODING_TYPE_PROPERTY = "encoding";

--- a/tools/tools-core/src/main/java/org/wso2/carbon/tools/spi/ICFProviderTool.java
+++ b/tools/tools-core/src/main/java/org/wso2/carbon/tools/spi/ICFProviderTool.java
@@ -48,6 +48,7 @@ import javax.tools.JavaCompiler;
 import javax.tools.ToolProvider;
 
 import static org.wso2.carbon.tools.Constants.BUNDLE_ACTIVATOR;
+import static org.wso2.carbon.tools.Constants.ECLIPSE_OSGI_VERSION;
 import static org.wso2.carbon.tools.Constants.JAR_MANIFEST_FOLDER;
 import static org.wso2.carbon.tools.Constants.MANIFEST_FILE_NAME;
 
@@ -102,8 +103,7 @@ public class ICFProviderTool implements CarbonTool {
         Path destination = Paths.get(toolArgs[2]);
         String osgiJar = (toolArgs.length == 4 && !toolArgs[3].isEmpty()) ? toolArgs[3] :
                          Paths.get(System.getProperty("carbon.home"), "wso2", "lib", "plugins",
-                                   "org.eclipse.osgi_3.14.0").toString();
-
+                                 ECLIPSE_OSGI_VERSION).toString();
         Path fileName = jarFile.getFileName();
         if (fileName == null) {
             return;

--- a/tools/tools-core/src/main/java/org/wso2/carbon/tools/spi/NativeLibraryProvider.java
+++ b/tools/tools-core/src/main/java/org/wso2/carbon/tools/spi/NativeLibraryProvider.java
@@ -48,6 +48,7 @@ import javax.tools.JavaCompiler;
 import javax.tools.ToolProvider;
 
 import static org.wso2.carbon.tools.Constants.BUNDLE_ACTIVATOR;
+import static org.wso2.carbon.tools.Constants.ECLIPSE_OSGI_VERSION;
 import static org.wso2.carbon.tools.Constants.JAR_MANIFEST_FOLDER;
 import static org.wso2.carbon.tools.Constants.MANIFEST_FILE_NAME;
 
@@ -97,7 +98,7 @@ public class NativeLibraryProvider implements CarbonTool {
         String nativeLibString = toolArgs[2];
         String osgiJar = (toolArgs.length == 4 && !toolArgs[3].isEmpty()) ? toolArgs[3] :
                 Paths.get(System.getProperty("carbon.home"), "wso2", "lib", "plugins",
-                        "org.eclipse.osgi_3.14.0").toString();
+                        ECLIPSE_OSGI_VERSION).toString();
 
         Path fileName = jarFile.getFileName();
         if (fileName == null) {

--- a/tools/tools-core/src/main/java/org/wso2/carbon/tools/spi/SPIProviderTool.java
+++ b/tools/tools-core/src/main/java/org/wso2/carbon/tools/spi/SPIProviderTool.java
@@ -47,6 +47,7 @@ import javax.tools.JavaCompiler;
 import javax.tools.ToolProvider;
 
 import static org.wso2.carbon.tools.Constants.BUNDLE_ACTIVATOR;
+import static org.wso2.carbon.tools.Constants.ECLIPSE_OSGI_VERSION;
 import static org.wso2.carbon.tools.Constants.JAR_MANIFEST_FOLDER;
 import static org.wso2.carbon.tools.Constants.MANIFEST_FILE_NAME;
 
@@ -100,7 +101,7 @@ public class SPIProviderTool implements CarbonTool {
         Path destination = Paths.get(toolArgs[3]);
         String osgiJar = (toolArgs.length == 5 && !toolArgs[4].isEmpty()) ? toolArgs[4] :
                 Paths.get(System.getProperty("carbon.home"), "wso2", "lib", "plugins",
-                        "org.eclipse.osgi_3.14.0.jar").toString();
+                        ECLIPSE_OSGI_VERSION).toString();
 
         Path fileName = jarFile.getFileName();
         if (fileName == null) {

--- a/tools/tools-core/src/test/java/org/wso2/carbon/tools/spi/ICFProviderToolTest.java
+++ b/tools/tools-core/src/test/java/org/wso2/carbon/tools/spi/ICFProviderToolTest.java
@@ -29,6 +29,7 @@ import java.util.jar.JarFile;
 import static org.testng.AssertJUnit.assertEquals;
 import static org.testng.AssertJUnit.assertNotNull;
 import static org.testng.AssertJUnit.assertTrue;
+import static org.wso2.carbon.tools.Constants.ECLIPSE_OSGI_VERSION;
 
 /**
  * Test class for {@link ICFProviderTool}.
@@ -52,7 +53,7 @@ public class ICFProviderToolTest {
         String spiImpl = "org.wso2.carbon.impl.TestSPIImpl";
         icfProviderTool.execute(spiImpl, sampleJARFile.toString(), converterTestResources.toString(),
                                 converterTestResources.getParent().resolve("lib")
-                                                      .resolve("org.eclipse.osgi_3.14.0.v20190517-1309.jar")
+                                                      .resolve(ECLIPSE_OSGI_VERSION)
                                                       .toString());
 
         String jarFileName = sampleJARFile.getFileName().toString();

